go.mod

```diff
@@ -6,7 +6,7 @@ require (
 	github.com/99designs/keyring v1.2.2
 	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.4.0
 	github.com/Azure/azure-sdk-for-go/sdk/storage/azblob v1.0.0
-	github.com/apache/arrow/go/v12 v12.0.1
+	github.com/apache/arrow/go/v14 v14.0.1
```

Comment:
Current version is 14.0.2, can we use it?

Reply:
there's still a couple of days to do the modification before January release change freeze hits, if we want to get this out with the Jan release

Reply:
@sfc-gh-pfus @sfc-gh-dheyman-1 would need your opinion on this because we seem to be using Arrow in the imports like https://github.com/snowflakedb/gosnowflake/blob/v1.7.1/arrow_chunk.go#L10-L12 (did not do an extensive search where it is referenced, this is just an example), perhaps all the imports needs to be changed to v14 as well?